src/coreclr/crossgen-corelib.proj

```diff
@@ -6,6 +6,7 @@
     <ItemGroup>
       <_CoreClrBuildArg Condition="'$(TargetArchitecture)' != ''" Include="-$(TargetArchitecture)" />
       <_CoreClrBuildArg Include="-$(Configuration.ToLower())" />
+      <_CoreClrBuildArg Condition="'$(TargetOS)' != '' and '$(TargetOS)' != '$(HostOS)'" Include="-os $(TargetOS) " />
```

Comment:
This will need to change.  What you want may be...  `'$(BuildOS)' != '$(TargetOS)'`, but we don't really have much code around `BuildOS` yet.